drivers/sensor/tmp108/tmp108.c

```diff
@@ -0,0 +1,406 @@
+int tmp108_reg_read(const struct device *dev, uint8_t reg, uint16_t *val)
+{
+	const struct tmp108_config *cfg = dev->config;
+	int result = 0;
+
+	result = i2c_burst_read_dt(&cfg->i2c_spec, reg, (uint8_t *) val, 2);
+
+	if (result < 0) {
+		return result;
+	}
+
+	*val = sys_be16_to_cpu(*val);
+
+	return 0;
+}
+
+int tmp108_reg_write(const struct device *dev, uint8_t reg, uint16_t val)
+{
+	const struct tmp108_config *cfg = dev->config;
+
+	uint8_t tx_buf[3] = { reg, val >> 8, val & 0xFF };
+
+	int result = i2c_write_dt(&cfg->i2c_spec, tx_buf, sizeof(tx_buf));
+
+	if (result < 0) {
+		return result;
+	}
+
+	return 0;
+}
+
+int tmp108_write_config(const struct device *dev, uint16_t mask, uint16_t conf)
+{
+	uint16_t config = 0;
+	int result = 0;
+
+	result = tmp108_reg_read(dev, TI_TMP108_REG_CONF, &config);
+
+	if (result < 0) {
+		return result;
+	}
+
+	config &= mask;
+	config |= conf;
+
+	result = tmp108_reg_write(dev, TI_TMP108_REG_CONF, config);
+
+	if (result < 0) {
+		return result;
+	}
+
+	return 0;
+}
+
+int ti_tmp108_read_temp(const struct device *dev)
+{
+	struct tmp108_data *drv_data = dev->data;
+	int result = 0;
+
+	/* clear previous temperature readings */
+
+	drv_data->sample = 0U;
+
+	/* Get the most recent temperature measurement */
+
+	result = tmp108_reg_read(dev, TI_TMP108_REG_TEMP, &drv_data->sample);
+
+	if (result < 0) {
+		return result;
+	}
+
+	return 0;
+}
+
+static int tmp108_sample_fetch(const struct device *dev,
+							enum sensor_channel chan)
+{
+	struct tmp108_data *drv_data = dev->data;
+	int result = 0;
+
+	/* If one shot mode is set, query chip for reading
+	 * reading should be finished 30 ms later
+	 */
+
+	if (drv_data->one_shot_mode == true) {
+
+		result = tmp108_write_config(dev,
+					TI_TMP108_MODE_MASK,
+					TI_TMP108_MODE_ONE_SHOT);
+
+		if (result < 0) {
+			return result;
+		}
+
+		/* Schedule read to start in 30 ms if mode change was successful
+		 * the typical wakeup time given in the data sheet is 27 ms
+		 */
+
+		result = k_work_schedule(&drv_data->scheduled_work,
+						K_MSEC(TMP108_WAKEUP_TIME_IN_MS));
+
+		if (result < 0) {
+			return result;
+		}
+
+		return 0;
+	}
+
+	result =  ti_tmp108_read_temp(dev);
+
+	if (result < 0) {
+		return result;
+	}
+
+	return 0;
+}
+
+static int tmp108_channel_get(const struct device *dev,
+				  enum sensor_channel chan,
+				  struct sensor_value *val)
+{
+	struct tmp108_data *drv_data = dev->data;
+	int32_t uval;
+
+	if (chan != SENSOR_CHAN_AMBIENT_TEMP &&
+		chan != SENSOR_CHAN_ALL) {
+		return -ENOTSUP;
+	}
+
+	uval = (int32_t)(drv_data->sample  >> 4) * TMP108_TEMP_MULTIPLIER;
+	val->val1 = uval / 1000000;
+	val->val2 = uval % 1000000;
+
+	return 0;
+}
+
+static int tmp108_attr_get(const struct device *dev,
+			   enum sensor_channel chan,
+			   enum sensor_attribute attr,
+			   struct sensor_value *val)
+{
+	int result = 0;
+
+	if (chan != SENSOR_CHAN_AMBIENT_TEMP &&
+		chan != SENSOR_CHAN_ALL) {
+		return -ENOTSUP;
+	}
+
+	switch ((int) attr) {
```

Comment:
why `int` cast here?

Reply:
I'm not sure, this might have been left over from testing.

Reply:
since this includes public attributes that are not listed in the sensor attr enum, I am getting warnings, if I don't cast this as an int example:

```
/home/jimmy/zephyrupstream/zephyr/drivers/sensor/tmp108/tmp108.c:271:2: warning: case value '15' not in enumerated type 'enum sensor_attribute' [-Wswitch]
  271 |  case SENSOR_ATTR_SHUTDOWN_MODE:

```